imports/lib/models/Base.ts

```diff
@@ -20,6 +20,11 @@ export type FindOptions = FindOneOptions & {
   limit?: number;
 }
 
+type SelectorToResultType<T, Selector extends FindSelector<T>> =
+  Selector extends string ? T & { _id: Selector } :
+  Selector extends Mongo.ObjectID ? T & { _id: Selector } :
+  T & { [K in keyof Selector & keyof T]: Selector[K] extends T[K] ? Selector[K] : never };
```

Comment:
Ah, I think I get it -- the string/ObjectID cases seem straightforward, and for this case: if we see a key that matches a key known to be in the model type `T`, then we propagate that as a constraint, and if it's not there (like `$in` or `$and` or potentially `arrayName.0` or other things that are permitted in `Mongo.Selector<T>` but aren't keys of `T`) then that adds no additional constraint.

Clever!